src/cargo/sources/registry/mod.rs

```diff
@@ -856,7 +856,13 @@ impl<'gctx> Source for RegistrySource<'gctx> {
                     }
                     any_pending |= self
                         .index
-                        .query_inner(name_permutation, &req, &mut *self.ops, f)?
+                        .query_inner(name_permutation, &req, &mut *self.ops, &mut |s| {
+                            if !s.is_yanked() {
+                                f(s);
+                            } else if kind == QueryKind::Alternatives {
+                                f(s);
+                            }
```

Comment:
I know at this moment `Alternatives` is for resolve to give a better suggestion, and `Normalized` is for `cargo add` to auto-normalize stuff. However, it is not clear why `Alternatives` deserves a special case here. To maximize the possibility to find an alternative to suggest?

Reply:
Talked offline. This is more a path toward a bigger overhual. Since the change is small and well-understood, resolve this thread.